### DISCUSSION TITLE
Is it possible to randomize choice order?

### DISCUSSION THREAD

**Discussion by aumusic** [dialogic-godot/dialogic#1737] [Q&A]
Is it possible to randomize the order of displayed choices within Dialogic? I was first given this idea by Ink (where it is simple to do), and the documentation for Dialogic 1.5 does show that it is possible to have a random word selected at a given point (or, I would imagine, a string of words if using variables) with a parameter like [word1,word2,word3] (though that seems to either be deprecated or use a different command set in Dialogic 2.0). The problem with going that route, it seems, is that all you'd be able to do is randomize the text within the choice, since each choice lead to fixed dialogue branches and/or fixed jumps (which makes total sense to do).

Does anyone have any thoughts on how this might be accomplished?

Thanks in advance

**Comment by Jowan-Spooner** [comment]
Hey @aumusic,
there is currently no such functionality. In 2.x this would probably be easy to implement, however I'm wondering where we would put this option. 
Three options I could see:
- a global dialogic setting could enable this for all questions
- a text-effect/modifier could do this (something like [randomize-choices])
- a setting on the choices. Obviously choices might be the best place, but it feels strange because: would it have to be enabled on all choices? or just one? idk.

But I like the idea.

**Comment by aumusic** [comment]
@Jowan-Spooner

Thank you for your response on this. I would personally think that the best approach would be the third one, since it could potentially be applied to individual choices (for example, randomizing the order of the first three choices, but keeping the fourth the same). Speaking from my own perspective, while I can't imagine doing that, I would personally like to have the ability to turn randomization on and off as needed (in my case, I'd like to randomize the order of most of the choices, but in a few places, keep them in a particular order).

Best regards
